Add KBI0010 on cloning from adjusted-mode checkout (#38)
psychoinformatics-de/knowledge-base: kbi/0010/index.rst

.. index::
   double: datalad; clone; adjusted mode

.. highlight:: console

KBI0010: Cloning a dataset that exists in the form of an adjusted mode checkout
===============================================================================

:authors: Adina Wagner <adina.wagner@t-online.de>

> :discussion: <link>

Author: ```suggestion
:discussion: https://github.com/psychoinformatics-de/knowledge-base/pull/38
```

> :keywords: adjusted mode, clone, cross-platform, Windows

Reviewer: ```suggestion
:keywords: adjusted mode, clone, cross-platform, Windows
:software-versions: datalad_0.18.3, git-annex_10.20221003
```

    [INFO   ] Detected a crippled filesystem.
    [INFO   ] Entering an adjusted branch where files are unlocked as this filesystem does not support locked files.
    [INFO   ] Switched to branch 'adjusted/master(unlocked)'
    create(ok): /media/mih/Samsung_T5/tmp/onwindows (dataset)

To make the demo a bit more life-like, we save a file into that dataset::

   $ datalad run -d /media/mih/Samsung_T5/tmp/onwindows 'dd if=/dev/urandom bs=1M count=10 of={dspath}/big.dat'
   [INFO   ] == Command start (output follows) =====
   10+0 records in
   10+0 records out
   10485760 bytes (10 MB, 10 MiB) copied, 0.0254713 s, 412 MB/s
   [INFO   ] == Command exit (modification check follows) =====
   run(ok): /media/mih/Samsung_T5/tmp/onwindows (dataset) [dd if=/dev/urandom bs=1M count=10 of=/me...]
   add(ok): big.dat (file)
   save(ok): . (dataset)

Because this dataset is on a "crippled filesystem", git-annex operates in
`adjusted mode`_.
This means, among other things, that the files in the worktree are actual files
instead of symlinks, and that the dataset has an "adjusted/<branch>(unlocked)"
version of every branch::

    $ ls -l /media/mih/Samsung_T5/tmp/onwindows

>     .rwxr-xr-x 10M mih mih 14 Apr 11:19  big.dat

Reviewer: ```suggestion
    .rwxr-xr-x 10M mih mih 14 Apr 11:19 big.dat
```

Remove unicode marker.

When cloning this adjusted-mode checkout onto a Linux system, DataLad is not able
to establish a proper clone.

> The clone will look like a dataset in adjust-mode, but it is not.

Reviewer: ```suggestion
The clone will look like a dataset in adjust-mode, although it is not.
```

Putting the clone into adjusted-mode via `git-annex adjust --unlock` also does
not yield a working dataset.

> To make things work, we simply remove the adjustment by checking out the regular
> branch (usually, it would be called ``master`` or ``main``) and killing the
> adjusted one::

Reviewer: ```suggestion
To make things work, we must remove the adjustment by checking out the regular
branch (usually, it would be called ``master`` or ``main``) and removing the
adjusted one::
```
